news/2025/2025-08-25-performance-points-star-rating-updates.md

```diff
@@ -0,0 +1,238 @@
+
+This is how the rhythm penalty looks on [MAX - Checklist (feat. Chromeo) [123 // abc]](https://osu.ppy.sh/beatmapsets/2155487#taiko/4543374) (a significantly rhythmically complex beatmap):
+
+![](/wiki/shared/news/2025-08-25-performance-points-star-rating-updates/taiko-rhythm-penalty.png)
+
+### Length bonus revamp
+
+A [change](https://github.com/ppy/osu/pull/33582) by [rloseise](https://osu.ppy.sh/users/6793778) was made in order to entirely revamp the length bonus in PP.
+
+Previously, the length bonus was applied based on the number of objects in the map and would reach its maximum bonus at 1500 objects. Length bonus now scales by a weighted object count depending on how consistently difficult the beatmap is, gradually converging towards a maximum bonus rather than ending abruptly:
+
+![](/wiki/shared/news/2025-08-25-performance-points-star-rating-updates/taiko-length-bonus.png)
+
+### Miss penalty changes
+
+A [change](https://github.com/ppy/osu/pull/33409) by [rloseise](https://osu.ppy.sh/users/6793778) was made in order to make the miss penalty more difficulty-aware.
+
+The new miss penalty is *also* scaled by a weighted object count depending on how consistently difficult the beatmap is, meaning frequent misses on less consistent beatmaps are punished more than misses on more consistent beatmaps:
+
+![](/wiki/shared/news/2025-08-25-performance-points-star-rating-updates/taiko-miss-penalty.png)
+
+### Rhythm object-snapping fix
+
+A [change](https://github.com/ppy/osu/pull/33403) by [ltca](https://osu.ppy.sh/users/11475208) and [BabySnakes](https://osu.ppy.sh/users/4669728) was made in order to fix an issue with "unsnapped" objects unfairly buffing rhythm difficulty.
+
+Prior to this fix, objects placed 1-5ms off of their intended timing would be treated as entirely new rhythms. These patterns would be played identically, however the system incorrectly saw these as constant, complex rhythm changes and would inflate star rating as a result. In some extreme cases, this caused inflation upwards of 2*.
+
+This fix works by normalising times between objects within a certain tolerance, meaning that 1ms unsnaps will be normalised to the same time and treated as the same rhythm. For maps not abusing this problem, this change has minimal effects.
+
+### Colour penalty fix
+
+A [change](https://github.com/ppy/osu/pull/33641) by [ltca](https://osu.ppy.sh/users/11475208) was made in order to fix a that caused patterns to unfairly avoid colour penalties.
+
+Colour difficulty contains a penalty for long patterns with no rhythm changes, but this penalty would only apply if the pattern's rhythm was *exactly* the same. This was addressed by looking back at previous notes and calculating an average, so that slightly missnapped notes are still correctly penalised.
```

Comment:
aight, now what's a "colour penalty"

```suggestion
The part of the PP system that evaluates colour difficulty has a penalty for long patterns with no rhythm changes. Before this update, there was an issue ([fixed](https://github.com/ppy/osu/pull/33641) by [ltca](https://osu.ppy.sh/users/11475208)) that prevented some patterns from receiving this penalty where they should have received it.

The issue was with the "no rhythm changes" part, that it had no tolerance for slightly missnapped notes. This was addressed by looking back at previous notes and calculating an average, so that the slightly missnapped notes are still correctly penalised.
```